Challange-1-iOS/Challange-1-iOS/Presentation/MainScreen/MainViewController.swift

```diff
@@ -2,8 +2,8 @@ import UIKit
 import Alamofire
 import CoreData
 
-class MainViewController: BaseGenericViewController<MainView>, Coordinating {
-    var coordinator: Coordinator?
+class MainViewController: BaseGenericViewController<MainView> {
+    public weak var delegate: MainViewControllerDelegate?
```

Comment:
why `public`?

Reply:
Don't need to be public, we can keep the variable as default (internal)